test/unit/rust_analyzer/rust_analyzer_test.bzl

```diff
@@ -0,0 +1,49 @@
+"""Unittests for rust rules."""
+
+load("@bazel_skylib//lib:unittest.bzl", "analysistest", "asserts")
+load("//rust:defs.bzl", "rust_analyzer", "rust_library")
+
+def _rust_analyzer_hello_world_test_impl(ctx):
```

Comment:
General question: What is the value in unit tests like these? I've always felt they were a bit tedious to maintain and didn't offer anything more than what an integration test would.

Reply:
To be honest, this test is there only because I started with the unittest and then realized I need the integration test. I probably should delete the unit test for the reasons you mention, and I will, unless there's a value in having a simple example of using Starlark unit testing, and a place where we can add future rust analyzer unit tests.

Reply:
I've not found a place where I value the unittest (when there's an integration test). I think integration tests exercise far more functionality as well as the important parts of a rule. But maybe I should care more about how many actions are in my rules?

Reply:
In general I reach for unit tests whenever I can. I find it more convenient to assert on registered actions, command lines, inputs, that sorts of things. 

I don't see a value in asserting how many actions are registered by a rule though, that was really just a hello world test and i wanted to assert something. I deleted the test now, it really doesn't pull its weight.